src/Uno.UI/UI/Xaml/Controls/ContentDialog/ContentDialog.Visuals.cs

```diff
@@ -0,0 +1,402 @@
+#nullable enable
+
+//// Copyright (c) Microsoft Corporation. All rights reserved.
+//// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using Uno.Extensions;
+using Windows.Foundation;
+using Windows.UI.ViewManagement;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Animation;
+
+namespace Windows.UI.Xaml.Controls
+{
+	partial class ContentDialog
+	{
+		const double ContentDialog_SIP_Bottom_Margin = 12.0;
+
+		#region .h
+		enum PlacementMode
+		{
+			Undetermined,
+			EntireControlInPopup,
+			TransplantedRootInPopup,
+			InPlace
+		};
+
+		PlacementMode m_placementMode = PlacementMode.Undetermined;
+		bool m_hideInProgress = false;
+		bool m_hasPreparedContent = false;
+		bool m_isShowing = false;
+		Storyboard? m_layoutAdjustmentsForInputPaneStoryboard;
+
+		double m_dialogMinHeight = 0;
+		#endregion
+
+		private protected override void ChangeVisualState(bool useTransitions)
+		{
+			base.ChangeVisualState(useTransitions);
+
+			bool fullSizeDesired = FullSizeDesired;
+
+			{
+				// ButtonsVisibilityStates
+				{
+					var primaryText = PrimaryButtonText;
+
+					var secondaryText = SecondaryButtonText;
+
+					var closeText = CloseButtonText;
+
+					bool hasPrimary = !primaryText.IsNullOrEmpty();
+					bool hasSecondary = !secondaryText.IsNullOrEmpty();
+					bool hasClose = !closeText.IsNullOrEmpty();
+
+					var buttonVisibilityState = "NoneVisible";
+					if (hasPrimary && hasSecondary && hasClose)
+					{
+						buttonVisibilityState = "AllVisible";
+					}
+					else if (hasPrimary && hasSecondary)
+					{
+						buttonVisibilityState = "PrimaryAndSecondaryVisible";
+					}
+					else if (hasPrimary && hasClose)
+					{
+						buttonVisibilityState = "PrimaryAndCloseVisible";
+					}
+					else if (hasSecondary && hasClose)
+					{
+						buttonVisibilityState = "SecondaryAndCloseVisible";
+					}
+					else if (hasPrimary)
+					{
+						buttonVisibilityState = "PrimaryVisible";
+					}
+					else if (hasSecondary)
+					{
+						buttonVisibilityState = "SecondaryVisible";
+					}
+					else if (hasClose)
+					{
+						buttonVisibilityState = "CloseVisible";
+					}
+
+					GoToState(useTransitions, buttonVisibilityState);
+				}
+
+				// DefaultButtonStates
+				{
+					var defaultButtonState = "NoDefaultButton";
+
+					var defaultButton = DefaultButton;
+
+					if (defaultButton != ContentDialogButton.None)
+					{
+						// UNO TODO
+						//ctl::ComPtr<DependencyObject> focusedElement;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6389897)